sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiveLinkHandler.java

```diff
@@ -17,28 +19,31 @@
 
 public class ReceiveLinkHandler extends LinkHandler {
     private final String linkName;
+    private final Function<Delivery, Message> deliveryDecoder;
     private AtomicBoolean isFirstResponse = new AtomicBoolean(true);
-    private final DirectProcessor<Delivery> deliveries;
-    private FluxSink<Delivery> deliverySink;
+    private final DirectProcessor<Message> messages;
+    private FluxSink<Message> messageSink;
 
-    public ReceiveLinkHandler(String connectionId, String hostname, String linkName, String entityPath) {
+    public ReceiveLinkHandler(String connectionId, String hostname, String linkName, String entityPath,
+        Function<Delivery, Message> deliveryDecoder) {
         super(connectionId, hostname, entityPath, new ClientLogger(ReceiveLinkHandler.class));
-        this.deliveries = DirectProcessor.create();
-        this.deliverySink = deliveries.sink(FluxSink.OverflowStrategy.BUFFER);
+        this.messages = DirectProcessor.create();
+        this.messageSink = messages.sink(FluxSink.OverflowStrategy.BUFFER);
         this.linkName = linkName;
+        this.deliveryDecoder = deliveryDecoder;
     }
 
     public String getLinkName() {
         return linkName;
     }
 
-    public Flux<Delivery> getDeliveredMessages() {
-        return deliveries;
+    public Flux<Message> getDeliveredMessages() {
```

Comment:
I was curious why we do decoding here? Does this break service bus? iirc, SB keeps track of deliveries when they are not in RECEIVE and delete mode to understand which ones to hold onto.